### PR TITLE
[TVMScript] Migrate More to TVMScripr Printer

### PR DESCRIPTION
**Pull request by junrushao**
This PR gradually migrates more pieces of the default printing to
TVMScript printer for TIR. Details:
- Introduced a method `AsLegacyRepr` which preserves existing
`AsRepr` provided by `ReprPrinter`, so that the legacy behavior
could be 100% preserved.
- Introduced `Script` method to `IRModule`, `PrimFunc`, `tir.Stmt`,
`tir.PrimExpr`. The `script` method exists in python side before,
and this PR introduced them to C++ to be consistent.
- Replace TIR's `PrettyPrint` to `operator <<` that is provided by
the new `ReprPrinter`, which outputs in TVMScript format by default.
`PrettyPrint` on Relay is all preserved for backward compatibility.

### CODE REVIEW

File: tests/python/unittest/test_meta_schedule_schedule_rule_mlt.py [hunk 1]
```diff
@@ -798,4 +885,5 @@ def max_pool_blocked_compute(height, width, channel):
 
 
 if __name__ == "__main__":
-    tvm.testing.main()
+    # tvm.testing.main()
+    test_cache_read_specify_consumer()
```
Comment (MasterJH5574) [inline tests/python/unittest/test_meta_schedule_schedule_rule_mlt.py:889]: Please revert this change :-)

**Review by MasterJH5574 — APPROVED** [review]
LGTM. Thanks for the migration. Just a few nits

File: include/tvm/script/printer/ir_docsifier_functor.h [hunk 2]
```diff
@@ -69,6 +69,9 @@ class IRDocsifierFunctor {
     if ((pf = LookupDispatchTable("", type_index)) != nullptr) {
       return (*pf)(obj, args...);
     }
+    LOG(WARNING) << "ObjectFunctor calls un-registered function on type: "
+                 << runtime::Object::TypeIndex2Key(type_index) << " (token: " << token << ")"
+                 << ". ObjectType: " << obj->GetTypeKey() << ". Object: " << obj;
```
[the remark this answers is not in the document]
Comment (junrushao) [reply include/tvm/script/printer/ir_docsifier_functor.h:74]: This is used to provide helpful debug information when there is outer logic that accidentally catches the error being thrown out

File: tests/python/unittest/test_meta_schedule_schedule_rule_mlt.py (hunk 1 above)
Comment (junrushao) [reply tests/python/unittest/test_meta_schedule_schedule_rule_mlt.py:889]: Oh ooops! Will do in a quick follow-up PR!

Comment (MasterJH5574) [reply tests/python/unittest/test_meta_schedule_schedule_rule_mlt.py:889]: Okay thanks
